### PR TITLE
[c++] Fix loop overflows

### PR DESCRIPTION
**Pull request by ara-ayvazyan**

### CODE REVIEW

**Comment by chwarr** [comment]
Don't worry about the "VSTS: Travis Ubuntu image - forks" check. That's me playing with more automation for the Travis CI images. But it's not working right now...

File: cpp/inc/bond/core/schema.h [hunk 1]
```diff
@@ -261,16 +265,26 @@ class InitSchemaDef
     template <typename T>
     uint16_t GetStructDef() const
     {
-        size_t n;
+        const auto& structs = _schema.structs;
+
+        BOOST_ASSERT(structs.size() <= (std::numeric_limits<uint16_t>::max)());
+
+        auto it = std::find_if(
+            std::cbegin(structs),
```
Comment (chwarr) [inline cpp/inc/bond/core/schema.h]: Apparently `cbegin` is [missing](https://travis-ci.org/Microsoft/bond/jobs/327429088) from the standard library we're using in Travis.

File: cpp/inc/bond/protocol/fast_binary.h [hunk 2]
```diff
@@ -373,7 +373,7 @@ class FastBinaryReader
         uint32_t size;
 
         ReadContainerBegin(size, element_type);
-        for (uint32_t i = 0; i < size; ++i)
+        for (int64_t i = 0; i < size; ++i)
```
Comment (chwarr) [inline cpp/inc/bond/protocol/fast_binary.h]: Don't the `SkipType(size_t)` functions also need checking?

Comment (ara-ayvazyan) [reply cpp/inc/bond/protocol/fast_binary.h]: I was planning to fix that separately. This change does not actually add any runtime checks.

**Pull request merged**